src/Files.Uwp/DataModels/SidebarPinnedModel.cs

```diff
@@ -259,19 +259,19 @@ public async Task AddItemToSidebarAsync(string path)
         {
             var item = await FilesystemTasks.Wrap(() => DrivesManager.GetRootFromPathAsync(path));
             var res = await FilesystemTasks.Wrap(() => StorageFileExtensions.DangerousGetFolderFromPathAsync(path, item));
-            if (res || (FilesystemResult)FolderHelpers.CheckFolderAccessWithWin32(path))
+            var lastItem = favoriteSection.ChildItems.LastOrDefault(x => x.ItemType == NavigationControlItemType.Location && !x.Path.Equals(CommonPaths.RecycleBinPath));
+            int insertIndex = lastItem != null ? favoriteSection.ChildItems.IndexOf(lastItem) + 1 : 0;
+            var locationItem = new LocationItem
```

Comment:
this is now needed even if the item is invalid, in order to display the item with inidicative icon.
Therefore was removed from the if-statement.